src/expr/impl/src/scalar/timestamptz.rs

```diff
@@ -46,22 +47,24 @@ pub fn timestamp_at_time_zone(input: Timestamp, time_zone: &str) -> Result<Times
     // https://www.postgresql.org/docs/current/datetime-invalid-input.html
     // Special cases:
     // * invalid time during daylight forward
-    //   * PostgreSQL uses UTC offset before the transition
-    //   * We report an error (FIXME)
+    //   * use UTC offset before the transition
     // * ambiguous time during daylight backward
-    //   * We follow PostgreSQL to use UTC offset after the transition
-    let instant_local = input
-        .0
-        .and_local_timezone(time_zone)
-        .latest()
-        .ok_or_else(|| ExprError::InvalidParam {
-            name: "local timestamp",
-            reason: format!(
-                "fail to interpret local timestamp \"{}\" in time zone \"{}\"",
-                input, time_zone
-            )
-            .into(),
-        })?;
+    //   * use UTC offset after the transition
+    let instant_local = match input.0.and_local_timezone(time_zone) {
+        LocalResult::Single(t) => t,
+        LocalResult::None => (input.0 + chrono::Duration::hours(1))
```

Comment:
FYI although the difference of 1-hour is very common, it is not always the case. As mention in `issue_12072.slt.part`, it could be half hour. Additionally, there may be one-off changes other than regular daylight saving, for example Singapore jumped from `1981-12-31 23:29:59` to `1982-01-01 00:00:00`. (`'1981-12-31 23:40'::timestamp at time zone 'Asia/Singapore'` in PostgreSQL corresponds to `1981-12-31T16:10:00Z`) The Singapore example is also half hour but such one-off change may be more arbitrary.

A real fix is to update the binary search logic inside `chrono` to locate time zone **offsets**. If a naive time does not match any entry, it shall return its preceding and following entry rather than `None`. So that we can pick the preceding entry to match PostgreSQL.

A quick hack may be to minus 3 hours in naive time first, do the timezone conversion, and add 3 hours back in the UTC timeline. This assumes jump forwards are less than 3 hours (IIRC there was 2-hour change) and there is a single change within this 3-hour window.

To clarify our word usage in this comment: `Asia/Singapore` is a **timezone**, while `UTC+07:30` and `UTC+08:00` are two **offsets** used by this **timezone**.